tests/integration/test_wpk/test_wpk_agent.py

```diff
@@ -0,0 +1,272 @@
+        'simulate_rollback': False,
+        'results': {
+            'upgrade_ok': False,
+            'error_message': 'err Could not chmod',
+            'receive_notification': False,
+        }
+    },
+    # 3. Simulate an interruption
+    {
+        'protocol': PROTOCOL,
+        'initial_version': _agent_version,
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': DEFAULT_UPGRADE_SCRIPT,
+        'chunk_size': 16384,
+        'simulate_interruption': True,
+        'simulate_rollback': False,
+        'results': {
+            'upgrade_ok': False,
+            'error_message': 'Request confirmation never arrived',
+            'receive_notification': False,
+        }
+    }
+]
+
+if _agent_version == 'v3.13.1':
+    test_metadata += [{
+        # 4. Simulate a rollback (v3.13.1)
+        'protocol': PROTOCOL,
+        'initial_version': 'v3.13.1',
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': DEFAULT_UPGRADE_SCRIPT,
+        'chunk_size': 16384,
+        'simulate_interruption': False,
+        'simulate_rollback': True,
+        'results': {
+            'upgrade_ok': True,
+            'result_code': 0,
+            'receive_notification': False,
+        }
+    }]
+elif _agent_version == 'v4.0.0':
+    test_metadata += [{
+        # 5. Simulate a rollback (v4.0.0)
+        'protocol': PROTOCOL,
+        'initial_version': 'v4.0.0',
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': DEFAULT_UPGRADE_SCRIPT,
+        'chunk_size': 16384,
+        'simulate_interruption': False,
+        'simulate_rollback': True,
+        'results': {
+            'upgrade_ok': True,
+            'result_code': 0,
+            'receive_notification': True,
+            'status': 'Failed',   
+        }
+    }]
+
+params = [
+    {
+        'CRYPTO': CRYPTO,
+        'SERVER_ADDRESS': SERVER_ADDRESS,
+        'REMOTED_PORT': 1514,
+        'PROTOCOL': PROTOCOL
+    } for x in range(0, len(test_metadata))
+]
+
+def load_tests(path):
+    """ Loads a yaml file from a path 
+    Retrun 
+    ----------
+    yaml structure
+    """
+    with open(path) as f:
+        return yaml.safe_load(f)
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_agent_conf.yaml')
+configurations = load_wazuh_configurations(configurations_path, __name__, params=params, metadata=test_metadata)
+
+# configurations = configurations[-1:]
+
+remoted_simulator = None 
+
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module"""
+    yield request.param
+
+@pytest.fixture(scope="function")
+def start_agent(request, get_configuration):
+    metadata = get_configuration['metadata']
+    authd_simulator = AuthdSimulator(server_address=SERVER_ADDRESS, enorllment_port=1515)
```

Comment:
Be careful, unexpected argument enorllment_port (enrollment_port). This should fail.